### ISSUE TITLE
Share scenario report

### ISSUE THREAD

**Issue by PVince81** [owncloud/diagnostics#34]
In many bug investigation cases (ex: shared storage hell series) we were missing the exact sharing scenario of a user to find out what is going wrong for example when mounting stuff.

We should implement a diagnostic command to report a sharing scenario up to one level for a specific user.

`occ diagnostics:sharingreport user1`

For example if we have "user3" shares "x" with "user2" and "user2" shares "y" with "user1".
When asking for a sharing report for user1, we get all received shares up to two levels:

- user1
    - receives "x" from user2
        - user2 receives "y" from user3

Additionally the command should report whether any of the mentioned users are zombies/disabled/...

So the sharing query needs to be done outside the API, maybe directly on the DB, to bypass the code that might exclude shares from the list due to disabled users.

@tomneedham as discussed yesterday evening

cc @cdamken @DeepDiver1975 @butonic @mrow4a

**Comment by Peter-Prochaska** [comment]
@PVince81 But this sharing report should never appear in a public issue tracker.

**Comment by tomneedham** [comment]
> @PVince81 But this sharing report should never appear in a public github repository.

You mean the output? Agreed - it is confidential information, including paths etc - but used to solve specific cases with admins, confidentially

**Comment by Peter-Prochaska** [comment]
@tomneedham Yes the output. But how can we make sure, that a user don't post it to a public repo?
